core/src/Cuda/Kokkos_Cuda_UniqueToken.hpp

```diff
@@ -110,9 +110,20 @@ class UniqueToken<Cuda, UniqueTokenScope::Global> {
 template <>
 class UniqueToken<Cuda, UniqueTokenScope::Instance>
     : public UniqueToken<Cuda, UniqueTokenScope::Global> {
+ private:
+  Kokkos::View<uint32_t*, ::Kokkos::CudaSpace> m_buffer_view;
+
  public:
   explicit UniqueToken(execution_space const& arg = execution_space())
       : UniqueToken<Cuda, UniqueTokenScope::Global>(arg) {}
+
+  UniqueToken(size_type max_size, execution_space const& = execution_space())
+      : m_buffer_view(
+            "UniqueToken::m_buffer_view",
+            ::Kokkos::Impl::concurrent_bitset::buffer_bound(max_size)) {
+    m_buffer = m_buffer_view.data();
+    m_count  = max_size;
```

Comment:
I don't think they can be, they're both members of the base class? And since m_buffer depends on m_buffer_view being allocated I don't think we could add a constructor to UniqueToken<Cuda, UniqueTokenScope::Global> to set it either?